src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] employeeSalaries = new int[names.length];
+
+        for (int i = 0; i < data.length; i++) {
+            String[] infoFromData = data[i].split(" ");
+            Employee employee = new Employee();
+            employee.setName(infoFromData[1]);
+
+            try {
+                if (infoFromData.length >= 4) {
+                    String dateString = infoFromData[0];
+                    if (!isDateInRange(dateString, dateFrom, dateTo)) {
+                        continue;
+                    }
+
+                    int hours = Integer.parseInt(infoFromData[2]);
+                    int salaryPerHour = Integer.parseInt(infoFromData[3]);
+                    int salary = hours * salaryPerHour;
+                    employee.setSalary(salary);
+
+                    for (int j = 0; j < names.length; j++) {
+                        if (names[j].equals(employee.getName())) {
+                            employeeSalaries[j] += salary;
+                            break;
+                        }
+                    }
+                } else {
+                    System.out.println("Wrong data format at index " + i);
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid number format at index " + i);
+            } catch (DateTimeException e1) {
+                System.out.println("Invalid date format at index " + i + ". Should be dd.mm.yyyy");
+            }
```

Comment:
remove the try-catch block. You don’t have any specific logic for these exceptions.